### PR TITLE
Inflation SLA - Subgraph Update

### PR DESCRIPTION
**Pull request by lucarlig**
This PR introduces an update to the DSLA Protocol subgraph schema and mapping, that accounts for the removal of the concept of "breached" contract.

## Warning
⚠️ We couldn't get IPFS to work locally on Linux @bussyjd 

## Validation

request:
```
{
  slas{
    id
    leverage
    deposits{
      type
      amount
      tokenAddress
    }
    SLIs{
      timestamp
      id
    }
  }
}
```
response:
```
{
  "data": {
    "slas": [
      {
        "SLIs": [],
        "deposits": [
          {
            "amount": "100000000000000000000000",
            "tokenAddress": "0xe78a0f7e598cc8b0bb87894b0f60dd2a88d6a8ab",
            "type": "user"
          },
          {
            "amount": "10000000000000000000000000",
            "tokenAddress": "0xe78a0f7e598cc8b0bb87894b0f60dd2a88d6a8ab",
            "type": "provider"
          }
        ],
        "id": "0x61d47da73822b4a77c4a9bae56ba25729669b9f1",
        "leverage": "20"
      },
      {
        "SLIs": [],
        "deposits": [
          {
            "amount": "10000000000000000000000000",
            "tokenAddress": "0xe78a0f7e598cc8b0bb87894b0f60dd2a88d6a8ab",
            "type": "provider"
          },
          {
            "amount": "200000000000000000000000",
            "tokenAddress": "0xe78a0f7e598cc8b0bb87894b0f60dd2a88d6a8ab",
            "type": "user"
          }
        ],
        "id": "0x97c849e6b83d6f9a0d4f10e439053c7ff6302e36",
        "leverage": "50"
      },
      {
        "SLIs": [],
        "deposits": [
          {
            "amount": "10000000000000000000000000",
            "tokenAddress": "0xe78a0f7e598cc8b0bb87894b0f60dd2a88d6a8ab",
            "type": "provider"
          },
          {
            "amount": "200000000000000000000000",
            "tokenAddress": "0xe78a0f7e598cc8b0bb87894b0f60dd2a88d6a8ab",
            "type": "user"
          }
        ],
        "id": "0xa91a06c80b27cffd2ce81af15558681b259714a9",
        "leverage": "50"
      }
    ]
  }
}
```

### CODE REVIEW

File: subgraph/networks/bsc.subgraph.yaml [hunk 1]
```diff
@@ -74,27 +74,47 @@ dataSources:
...
+      language: wasm/assemblyscript
+      entities:
+        - PeriodRegistry
+      abis:
+        - name: PeriodRegistry
+          file: ../abis/PeriodRegistry.json
+      eventHandlers:
+        - event: PeriodModified(uint8,uint256)
+          handler: handlePeriodModified
+      file: ../mapping.ts
```
Comment (wilhempujar) [inline subgraph/networks/bsc.subgraph.yaml]: Why is this back?
Let's review all yaml to make sure it's not.

File: services/graph-protocol/docker-compose.yaml [hunk 2]
```diff
@@ -15,16 +17,19 @@ services:
       postgres_user: graph-node
       postgres_pass: let-me-in
       postgres_db: graph-node
-      ipfs: 'http://host.docker.internal:5001'
+      ipfs: 'https://api.thegraph.com/ipfs/'
```
Comment (wilhempujar) [inline services/graph-protocol/docker-compose.yaml:20]: Is this the part where Docker IPFS isn't working properly on Linux?

File: subgraph/deploy-develop.sh [hunk 3]
```diff
@@ -1,4 +1,4 @@
-export IPFS_URL="http://localhost:5001"
+export DEVELOP_IPFS_URI="https://api.thegraph.com/ipfs/"
```
Comment (wilhempujar) [inline subgraph/deploy-develop.sh]: Okay. I'll leave @bussyjd comment the approach here, but It's contextually meaningful.

**Review by wilhempujar — CHANGES_REQUESTED** [review]
LGTM other than the branch name (not following conventions) + unused code that needs to be removed (PeriodRegistry handler configuration).

**Comment by wilhempujar** [comment]
Updated this PR metadata (title + description).

Rich templates could be useful from now on, we need to make sure that changes a crystal clear to our global audience.

File: services/graph-protocol/docker-compose.yaml (hunk 2 above)
Comment (lucarlig) [reply services/graph-protocol/docker-compose.yaml:20]: yes

File: subgraph/networks/bsc.subgraph.yaml (hunk 1 above)
Comment (lucarlig) [reply subgraph/networks/bsc.subgraph.yaml]: removed

File: subgraph/deploy-develop.sh (hunk 3 above)
Comment (bussyjd) [reply subgraph/deploy-develop.sh]: Already validated with @lucarlig :+1:

**Review by bussyjd — APPROVED** [review]
LGTM

**Comment by bussyjd** [comment]
@lucarlig local IPFS is fixed and period related tasks are reverted

**Pull request merged**